src/signal.rs

```diff
@@ -183,21 +159,30 @@ pub fn validate_signal_num(num: c_int, for_vcpu: bool) -> errno::Result<c_int> {
 /// use vmm_sys_util::signal::{register_signal_handler, SignalHandler};
 ///
 /// extern "C" fn handle_signal(_: c_int, _: *mut siginfo_t, _: *mut c_void) {}
-/// unsafe {
-///     register_signal_handler(0, SignalHandler::Siginfo(handle_signal), true, SA_SIGINFO);
-/// }
+/// register_signal_handler(0, handle_signal, true, SA_SIGINFO);
 /// ```
 
-pub unsafe fn register_signal_handler(
+pub fn register_signal_handler(
     num: c_int,
     handler: SignalHandler,
     for_vcpu: bool,
```

Comment:
I really think this `for_vcpu` is simply wrong. Like I said before, let's just remove it. You can do it in this PR or I can open a new one after, whichever you like.